v2/common/src/main/java/com/google/cloud/teleport/v2/transforms/ElasticsearchTransforms.java

```diff
@@ -0,0 +1,259 @@
+    String getTypeFnPath();
+
+    void setTypeFnPath(String typeFnPath);
+
+    @Description("Optional: Name of Javascript function to extract Type from document.")
+    String getTypeFnName();
+
+    void setTypeFnName(String typeFnName);
+  }
+
+  /**
+   * The {@link WriteToElasticsearch} class writes a {@link PCollection} of strings to Elasticsearch
+   * using the following options.
+   *
+   * <ul>
+   *   <li>{@link WriteToElasticsearchOptions#getNodeAddresses()} - comma separated list of nodes.
+   *   <li>{@link WriteToElasticsearchOptions#getIndex()} - index to output documents to.
+   *   <li>{@link WriteToElasticsearchOptions#getDocumentType()} - document type to write to.
+   *   <li>{@link WriteToElasticsearchOptions#getBatchSize()} - batch size in number of documents
+   *       (Default:1000).
+   *   <li>{@link WriteToElasticsearchOptions#getBatchSizeBytes()} - batch size in number of bytes
+   *       (Default:5000000).
+   *   <li>{@link WriteToElasticsearchOptions#getMaxRetryAttempts()} - optional: maximum retry
+   *       attempts for {@link ElasticsearchIO.RetryConfiguration}.
+   *   <li>{@link WriteToElasticsearchOptions#getMaxRetryDuration()} - optional: maximum retry
+   *       duration for {@link ElasticsearchIO.RetryConfiguration}.
+   *   <li>{@link WriteToElasticsearchOptions#getUsePartialUpdate()} - use partial updates instead
+   *       of insertions (Default: false).
+   * </ul>
+   */
+  @AutoValue
+  public abstract static class WriteToElasticsearch extends PTransform<PCollection<String>, PDone> {
+
+    public static Builder newBuilder() {
+      return new AutoValue_ElasticsearchTransforms_WriteToElasticsearch.Builder();
+    }
+
+    public abstract WriteToElasticsearchOptions options();
+
+    @Override
+    public PDone expand(PCollection<String> jsonStrings) {
+
+      ConnectionConfiguration config =
+          ConnectionConfiguration.create(
+              options().getNodeAddresses().split(","),
+              options().getIndex(),
+              options().getDocumentType());
+
+      RetryConfiguration retryConfig = null;
+      if (options().getMaxRetryAttempts() != null) {
+        retryConfig =
+            RetryConfiguration.create(
+                options().getMaxRetryAttempts(), getDuration(options().getMaxRetryDuration()));
+      }
+
+      ElasticsearchIO.Write write =
+          ElasticsearchIO.write()
+              .withConnectionConfiguration(config)
+              .withMaxBatchSize(options().getBatchSize())
+              .withMaxBatchSizeBytes(options().getBatchSizeBytes())
+              .withUsePartialUpdate(options().getUsePartialUpdate());
+
+      if (retryConfig != null) {
+        write.withRetryConfiguration(retryConfig);
+      }
+
+      return jsonStrings.apply(
+          "WriteDocuments",
+          write
+              .withIdFn(
+                  ValueExtractorFn.newBuilder()
```

Comment:
What happens when the ValueExtractorFn returns a null (i.e when no function is applied). What is the expected behavior in that case?

For this and below;.